src/App.tsx

```diff
@@ -1,24 +1,220 @@
-/* eslint-disable max-len */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable react-hooks/rules-of-hooks */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { UserWarning } from './UserWarning';
+import { Loader } from './components/Loader';
+import { FilterTypes } from './components/TodoFilter';
+import { Todos } from './components/Todos';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import {
+  deleteTodos, getTodos, patchTodos, postTodos,
+} from './api/todos';
 
-const USER_ID = 0;
+export const USER_ID = '10682';
+
+export enum TodoErros {
+  Add = ' Unable to add a todo',
+  Delete = 'Unable to delete a todo',
+  Update = 'Unable to update a todo',
+  ErrorTodo = 'Can not find todos',
+}
 
 export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [input, setInput] = useState<string>('');
+  const [filter, setFilter] = useState(FilterTypes.All);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string>('');
+  const [tempTodoId, setTempTodoId] = useState<number | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos(USER_ID).then(
+      fetchedTodos => {
+        setTodos(fetchedTodos as Todo[]);
+        setIsLoading(false);
+      },
+    ).catch(() => setError(TodoErros.ErrorTodo));
+  }, []);
+
+  const handleImputTodo = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setInput(e.target.value);
+  };
+
+  const handleAddTodo = (
+    e: React.FormEvent<HTMLFormElement>,
+  ) => {
+    e.preventDefault();
+    if (input.trim()) {
+      const newTodo: Omit<Todo, 'id'> = {
+        userId: Number(USER_ID),
+        title: input,
+        completed: false,
+      };
+
+      postTodos(USER_ID, newTodo).then((todo) => {
+        const receivedTodo = todo as Todo;
+
+        setTempTodoId(receivedTodo.id);
+
+        setTodos((prevTodos) => [...prevTodos, receivedTodo]);
+        setInput('');
+        setTempTodoId(null);
+
+        if (error && error === TodoErros.Add) {
+          setError('');
+        }
+      }).catch(() => setError(TodoErros.Add));
+    }
+  };
+
+  const handleRemoveTodo = (todoId: number) => {
```

Comment:
A lot of functionality is implemented inside the App component. Check if you can move some logic outside the component, or refactor it somehow to make the component more clear and maintainable.